src/IR/DesugarPattern.hs

```diff
@@ -157,7 +157,16 @@ desugarExpr (I.Match e arms t) = do
         )
         arms
     let armsForDesugar = map (first (: [])) arms'
-    desugarMatch [e] armsForDesugar (I.NoExpr t)
+    case e of
+        I.Var _ _ -> desugarMatch [e] armsForDesugar (I.NoExpr t)   -- TODO: add let alias
+        _         -> do
+            v <- freshVar
+            singleLet (I.VarId v) e
+                <$> desugarMatch [e] armsForDesugar (I.NoExpr t)
```

Comment:
I think a modification should be made here, please see the new commits.

`<$> desugarMatch [e] armsForDesugar (I.NoExpr t)` 

Should be

`<$> desugarMatch [I.Var var vt] armsForDesugar (I.NoExpr t)`

Reply:
agreed.